Add MoMo robot

manavthakkar opened the pull request:
This PR adds momo robot urdf along with the scad files for collision approximation. MoMo is a mecanum wheeled robot developed at the [Institut für Technische Logistik](https://www3.tuhh.de/itl/) - Hamburg, Germany. 

When i drag and drop the momo_urdf in this [urdf web viewer](https://gkjohnson.github.io/urdf-loaders/javascript/example/bundle/index.html), everything works perfectly, but when i use the onshape-to-robot-bullet, the wheels are missing.

![Screenshot from 2025-03-17 12-00-03](https://github.com/user-attachments/assets/a2526e75-6116-4e8c-b2c2-fc7de5e4d8d7)

![Screenshot from 2025-03-17 11-09-26](https://github.com/user-attachments/assets/6c19e4ef-9f7c-414e-a857-7ce1bc3e448d)

![Screenshot from 2025-03-17 11-09-43](https://github.com/user-attachments/assets/dda28c3c-0e22-4749-9537-f75e19e38edc)

manavthakkar committed:
Add MoMo urdf

manavthakkar committed:
Update README

Gregwar commented:
Hello,

Thanks for your PR!

Here are some remarks:

## There are unfixed parts. You used "groups" to attach x/y/z axises, which is not supported by onshape-to-robot.

Instead, you should consider one of the following:
    - Use the composite part "frame" (you have to click on the composite button to enable it, see below)
    - Simply add a mate connector called "frame_something", without using a separate body

![image](https://github.com/user-attachments/assets/bcc8c0aa-65fe-43aa-9bd4-a8761be5986d)

## The robot is not loading entirely in pyBullet, because of mesh sizes.

Here is what I changed:

- Set joint type to continuous
- Ignored inner parts and connectors
- Enabled meshes simplification and merge, with a limit to 1M

```json
{
    "url": "https://cad.onshape.com/documents/0f7ba08d49760d832652e76c/w/f802382a554ac4e5d967eb06/e/176c79e9883e5c00c7190ba1",
    "output_format": "urdf",
    "use_scads": true,
    "simplify_stls": true,
    "merge_stls": true,
    "max_stl_size": 1,

    "joint_properties": {
        "*wheel*": {
            "type": "continuous"
        }
    },

    "ignore": {
        "Aluminium_profile*": "all",
        "*Connector*": "all"
    }
}
```

![image](https://github.com/user-attachments/assets/9e71ab39-8277-486d-8017-292ead16a0bb)

## MuJoCo support

Here is a suitable configuration for MuJoCo

```python
{
    "url": "https://cad.onshape.com/documents/0f7ba08d49760d832652e76c/w/f802382a554ac4e5d967eb06/e/176c79e9883e5c00c7190ba1",
    "output_format": "mujoco",
    "use_scads": true,

    "joint_properties": {
        "*": {
                "actuated": false,
                "range": false
        },
        "*_wheel": {
           "actuated": true,
           "type": "velocity",
           "limits": [-16, 16],
           "kv": 5.0
        }
    }
}
```


https://github.com/user-attachments/assets/2176033a-2670-4682-add1-0750a7c0d6c6

manavthakkar committed:
Update config, README and add momo_ mujoco

manavthakkar commented:
Thanks for the feedback!

I added the mate connectors with name "frame_something", without using separate bodies, and updated the configuration file for the urdf, and added momo_mujoco with the configuration that you provided.
